gadopt/stokes_integrators.py

```diff
@@ -255,9 +243,25 @@ def __init__(
                     weak_bc[bc_type] = value
             self.weak_bcs[id] = weak_bc
 
+        u, p = fd.split(self.solution)
+        terms_kwargs = {"u": u, "p": p, "T": T, "mu": mu}
```

Comment:
As with the energy solver, I don't think it's made clear anywhere that this is the required set of functions for the particular set of equations.